easybuild/easyblocks/generic/bundle.py

```diff
@@ -124,8 +125,22 @@ def __init__(self, *args, **kwargs):
                 comp_cfg['name'] = comp_name
                 comp_cfg['version'] = comp_version
 
+                # The copy above may include unexpected settings for common values.
+                # In particular for a Pythonbundle we have seen a component inheriting
+                #  runtest = True
+                # which is not a valid value for many easyblocks.
+                # Reset runtest to the original default, if people want the test step
+                # they can set it explicitly (or it can be set by the component easyblock).
+                if comp_cfg._config['runtest'] != DEFAULT_CONFIG["runtest"]:
+                    self.log.warning(
+                        "Resetting runtest to default value for component easyblock "
+                        f"(from {comp_cfg._config['runtest']})."
+                        )
+                    comp_cfg["runtest"] = DEFAULT_CONFIG["runtest"][0]
+                    comp_cfg._config['runtest'] = DEFAULT_CONFIG["runtest"]
```

Comment:
The 2nd overwrites the first, doesn't it?

```suggestion
                    comp_cfg._config['runtest'] = DEFAULT_CONFIG["runtest"]
```